### PR TITLE
add create and inset composite modification network

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 1]
```diff
@@ -211,6 +213,13 @@ public ResponseEntity<UUID> createModificationInGroup(@RequestBody ModificationI
         return ResponseEntity.ok().body(networkModificationService.createModificationInGroup(modificationsInfos));
     }
 
+    @PostMapping(value = "/composite/network-modifications", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Create a composite modification")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The composite modification has been created")})
+    public ResponseEntity<UUID> createCompositeModification(@RequestBody ModificationInfos modificationInfos) {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:219]: we remove this specific endpoint nope ?

Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:219]: yes
I am working on it

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 2]
```diff
@@ -282,12 +282,49 @@ public Optional<NetworkModificationResult> duplicateModifications(UUID targetGro
         return applyModifications(networkUuid, variantId, reportInfos, modificationInfos);
     }
 
+    @Transactional
+    public Optional<NetworkModificationResult> insertModifications(UUID targetGroupUuid,
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java]: done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 3]
```diff
@@ -216,7 +216,7 @@ public ResponseEntity<UUID> createModificationInGroup(@RequestBody ModificationI
     @PostMapping(value = "/network-composite-modifications", consumes = MediaType.APPLICATION_JSON_VALUE)
     @Operation(summary = "Create a network composite modification")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The composite modification has been created")})
-    public ResponseEntity<UUID> createNetworkCompositeModification(@RequestBody ModificationInfos modificationInfos) {
+    public ResponseEntity<UUID> createNetworkCompositeModification(@RequestBody List<UUID> modificationInfos) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:219]: modificationUuids ?

**Review by SlimaneAmar — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/entities/CompositeModificationEntity.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+@NoArgsConstructor
+@Getter
+@Setter
+@Entity
+@Table(name = "composite_modification")
+public class CompositeModificationEntity extends ModificationEntity {
+
+    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
+    @JoinTable(
+            name = "compositeModificationList",
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/entities/CompositeModificationEntity.java:32]:
```suggestion
            name = "modifications",
```

**Review by etiennehomer — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/repositories/ModificationRepository.java [hunk 5]
```diff
@@ -35,12 +35,18 @@ public interface ModificationRepository extends JpaRepository<ModificationEntity
     @Query(value = "SELECT new ModificationEntity(m.id, m.type) FROM ModificationEntity m WHERE m.id IN (?1)")
     List<ModificationEntity> findMetadataIn(List<UUID> uuids);
 
+    @Query(value = "SELECT m FROM ModificationEntity m WHERE m.id IN (?1) ORDER BY m.modificationsOrder")
+    List<ModificationEntity> findAllByOrderId(List<UUID> uuids);
```
[the remark this answers is not in the document]
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/repositories/ModificationRepository.java:39]: or findAllByModificationsOrder ?

File: src/main/java/org/gridsuite/modification/server/entities/CompositeModificationEntity.java (hunk 4 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/modification/server/entities/CompositeModificationEntity.java:32]: compositeModificationSubModifications

**Review by etiennehomer — APPROVED** [review]

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**